### DISCUSSION TITLE
[Node Operator Question] Execution Client time to find peers (Nethermind)

### DISCUSSION THREAD

**Discussion by emlautarom1** [ethereum-optimism/developers#670] [Miscellaneous] [answered]
### Did you check the documentation and specs?

- [X] Yes

### Was there anything unclear or missing?

We would like to know what is the expected average time to find peers for the Execution Engine.

### Please select the type of request

- [ ] Bug Report
- [X] Feature Request

### Feature Description

At Nethermind we're currently improving our OP Stack support and for that we are constantly running and syncing Optimism nodes in both Mainnet and Sepolia. In our testing pipeline we found a lot of inconsistencies when it comes to syncing nodes from scratch, in particular when it comes to finding peers: it takes quite a bit of time to find them (ranging from 5 to 20 minutes), but we eventually find some.

We were unable to find any issues discussing how long does it take to find peers using op-geth, so it's unclear if these times are within reason or not.

One of our developers suggested to spin up some additional boot nodes to ensure that syncing is more stable for Nethermind nodes, either under an Optimism or Base grant, or under Nethermind administration and budget.

### Purpose and Benefits

For Nethermind it would greatly reduce testing time since we spend several hour syncing nodes every time we merge a PR or prepare a release. For other clients we expect that sync time would also be reduced in case these nodes are public.

### Relevant Context or Examples

For example, a recent node took ~7 minutes to find peers. In other test runs we've seen it take it up to 30 minutes, but results are quite inconsistent.

**Comment by hash1go** [comment]
Thank you for sharing. When I have the opportunity, I will relay this to the OP team.

**Reply by sbvegan** [reply, marked as answer]
Hey @emlautarom1, here's an invite to the [Optimism R&D server](https://discord.gg/ZtVzh8JZ). I recommend pinging this thread in there so you can have an engineer take a look

**Reply by emlautarom1** [reply]
Thanks. I'll forward the question to them.
